File: Data Structures & Algorithms/coin-change/submission-14.py
class Solution:
    def coinChange(self, coins: List[int], amount: int) -> int:
        if amount==0:
            return 0
        total=set()
        big=True
        for i in coins:
            if i<=amount:
                big=False
            if i==amount:
                return 1
            total.add((i,1))
        if big:
            return -1
        print(total)
        while True:
            temp=set()
            for val, num in total:
                tooBig=True
                for i in coins:
                    newVal=val+i
                    newNum=num+1
                    if newVal==amount:
                        return newNum
                    if newVal<=amount:
                        tooBig=False
                    temp.add((newVal,newNum))
            if tooBig:
                return -1
            total.update(temp)

File: Data Structures & Algorithms/climbing-stairs/submission-14.java
class Solution {
    public int climbStairs(int n) {
        if (n<2){
            return n;
        }
        int[] DP=new int[n];
        DP[0]=1;
        DP[1]=2;
        for(int i=2;i<DP.length;i++){
            DP[i]=DP[i-1]+DP[i-2];
        }
        return DP[DP.length-1];
    }
}
